services/language_translation/v2.js

```diff
@@ -21,358 +22,354 @@ module.exports = function (RED) {
+      var language_translation = watson.language_translation({
         username: username,
         password: password,
         version: 'v2'
       });
 
-      switch (action) {
-        case 'translate':
-          this.doTranslate(msg);
-          break;
-        case 'train':
-          this.doTrain(msg);
-          break;
-        case 'getstatus':
-          this.doGetStatus(msg, trainid);
-          break;
-        case 'delete':
-          this.doDelete(msg, trainid);
-          break;
-      }
-    });
-
-    this.doTranslate = function (msg) {
-      var domain = msg.domain || config.domain;
-
-      if (!domain) {
-        node.warn('Missing translation domain, message not translated');
+      var action = msg.action || config.action;
+      if (!action) {
+        node.warn('Missing action, please select one');
         node.send(msg);
         return;
       }
 
-      var srclang  = ''; 
-      var destlang = '';
-
-      switch (domain) {
-        case 'news':
-          srclang = config.srclangnews;
-
-          if (!srclang) {
-            node.warn('Missing source language, message not translated');
-            node.send(msg);
-            return;
-          }
-
-          destlang = config.destlangnews;
-
-          if (!destlang) {
-            node.warn('Missing target language, message not translated');
+      // We have credentials, and know the mode. Further required fields checks
+      // are specific to the requested action.
+      // The required fields are checked, before the relevant function is invoked.
+      switch (action) {
+        case 'translate':
+          var domain = msg.domain || config.domain;
+          if (!domain) {
+            node.warn('Missing translation domain, message not translated');
             node.send(msg);
             return;
           }
-          break;
-        case 'conversational':
-          srclang = config.srclangconversational;
-
+          var srclang = msg.srclang || config.srclang;
           if (!srclang) {
             node.warn('Missing source language, message not translated');
             node.send(msg);
             return;
           }
-
-          destlang = config.destlangconversational;
-
+          var destlang = msg.destlang || config.destlang;
           if (!destlang) {
             node.warn('Missing target language, message not translated');
             node.send(msg);
             return;
           }
-          break;
-        case 'patent':
-          srclang = config.srclangpatent;
-
-          if (!srclang) {
-            node.warn('Missing source language, message not translated');
-            node.send(msg);
-            return;
-          }
-
-          destlang = config.destlangpatent;
-
-          if (!destlang) {
-            node.warn('Missing target language, message not translated');
-            node.send(msg);
-            return;
+          var model_id = '';
+          model_id = srclang + '-' + destlang;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2613300751/issues/source?bid=3225115&fileBranchId=3311803#l321)